File: backend/src/main/java/com/spring/restaurant/model/Order.java
package com.spring.restaurant.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "orders",
        uniqueConstraints = {
                @UniqueConstraint(columnNames = "order_name")
        })
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "order_id")
    private int orderId;
    @Column(name = "order_name", unique = true, nullable = false)
    private String orderName;
    @Column(name = "order_price", nullable = false, precision = 10, scale = 2)
    private BigDecimal orderPrice;
    private String image;
    @Column(name = "order_description")
    private String orderDescription;

    @Column(name = "creation_date")
    @CreationTimestamp
    private LocalDate creationDate;

    @Column(name = "update_date")
    @UpdateTimestamp
    private LocalDateTime updateDate;


    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;


}

File: frontend/src/app/model/order.ts
import {Category} from './category';

export class Order {

  orderId : number;
  orderName : string;
  orderPrice : number;
  image : string;
  orderDescription : string;
  creationDate: string;
  updateDate: string;
  category: Category;


  constructor(orderId: number, orderName: string, orderPrice: number, image: string, orderDescription: string, creationDate: string, updateDate: string, category: Category) {
    this.orderId = orderId;
    this.orderName = orderName;
    this.orderPrice = orderPrice;
    this.image = image;
    this.orderDescription = orderDescription;
    this.creationDate = creationDate;
    this.updateDate = updateDate;
    this.category = category;
  }
}
